lib/libvmod_blob/hex.c

```diff
@@ -99,22 +99,35 @@ hex_encode(const enum encoding enc, const enum case_e kase,
 		*p++ = alphabet[in[i] & 0x0f];
 	}
 
+	/* XXX: the length arithmetics here are a bit complicated for such
+	 * a simple problem. It looks like the estimated encoded length
+	 * makes room for a null character, which makes sense for string
+	 * encoding. But null characters are set by the caller as documented.
+	 *
+	 * The result is a confusing length computation where:
+	 *
+	 *     hex_decode_l(hex_encode_l(len)) != len
+	 *
+	 * Needs a proper review of "half byte" decoding. Should it be
+	 * supported in the first place?
+	 */
```

Comment:
What's the alternative? Fail for hex encodings with an odd length? Hex strings like "123" have a common and intuitive interpretation (0x0123 == 291 decimal).